### DISCUSSION TITLE
Welcome to Ideas

### DISCUSSION THREAD

**Discussion by chipkent** [deephaven/deephaven-core#972] [Ideas]
Do you have ideas for how to improve Deephaven?  This is the place to discuss them!

**Comment by nbauernfeind** [comment]
As a developer, I would like richer integration with errors that hit the UI. I can already tell that we detect them enough to collapse them.

I suggest that we add a context menu if right clicked:
1. click copy to clipboard (can't tell you how many times that thing has collapsed on me when I was trying to highlight-to-copy; but I pull these into my IDE almost every time)
2. generate a URL to search github (discussions, issues, and/or PRs? this will be a nice way to offer easily found support documentation for common user errors)
3. open a github discussion and/or issue filled in with the error details (as a developer, I would love to have this also auto-populate a build-id of some sort)

What do you think @mofojed, is there some low hanging UX-fruit here?

**Reply by mofojed** [reply]
General comments: 
- For detecting when to collapse them, we're just checking if it's a multiline error and taking just the first line. So not doing anything fancy there. 
- For selecting, it looks like there's a bug there where it may sometimes collapse the error message when you don't want it to, I'll submit a fix for that.

The other 3 suggestions require overriding the native context menu, which would mean any browser extensions you have wouldn't work, eg. if you had something like this: https://chrome.google.com/webstore/detail/stackoverflow-right-click/gaebaakogoamheaakiiekhmceginhnma?hl=en.
I'm not sure we want to have our own context menu in this particular context. @dsmmcken may have an opinion.
